zen_document_parser/base.py

```diff
@@ -0,0 +1,231 @@
+from enum import IntEnum
+from pdfquery import PDFQuery
+from pdfminer.pdfdocument import PDFSyntaxError
+
+from zen_document_parser.exceptions import InvalidPDFError, UnknownVariantError
+
+
+class DocFieldType(IntEnum):
+    TEXT = 1
+    NUMBER = 2
+    CUSTOM = 3  # TODO: Forget this and have 'type' take a callable instead?
+
+
+class DocField(object):
+    def __init__(self, bbox, type=DocFieldType.TEXT, required=False,
+                 description=None):
+        self.bbox = bbox
+        self.type = type
+        self.required = required
+        self.description = description
+
+
+class DocVariantMeta(type):
```

Comment:
I am fine with metaclass approach for now and am closing this PR indicating that the code is fine.